org.eclipse.jdt.core/codeassist/org/eclipse/jdt/internal/codeassist/DOMCompletionEngineVariableDeclHandler.java

```diff
@@ -0,0 +1,40 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ * 
+ * Contributors:
+ *     Gayan Perera - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jdt.internal.codeassist;
+
+import java.util.List;
+
+import org.eclipse.jdt.core.dom.IVariableBinding;
+import org.eclipse.jdt.internal.codeassist.DOMCompletionEngine.Bindings;
+
+/**
+ * This class define methods which are used for handling dom based completions for variable declarations.
+ */
```

Comment:
@mickaelistria I made this class in the intention of adding more and more logic for resolving parameter names with different strategies and may be later add support for different kind of VariableDeclarations based operations to avoid make the DOMCompletionEngine bloated with lot of logic. Lets me know otherwise.

Reply:
I think a new class is fine, in any case, we can easily refactor if we change our mind later.